testing_mad.ipynb

```diff
@@ -2,130 +2,176 @@
  "cells": [
   {
    "cell_type": "code",
-   "execution_count": 6,
+   "execution_count": 2,
    "id": "6dece361-8765-4ba7-a49c-b165b86356ee",
    "metadata": {},
    "outputs": [],
    "source": [
     "import os\n",
-    "import pandas as pd"
+    "import pandas as pd\n",
+    "import unittest"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 16,
+   "execution_count": 7,
    "id": "874d6f28-222e-46be-bf26-613c793110e1",
    "metadata": {},
    "outputs": [],
    "source": [
     "# function to calculate median absolute deviation\n",
-    "def mad(x): \n",
-    "    abs(x - x.median(axis='columns', numeric_only=True)).median(axis='columns', numeric_only=True)"
+    "def mad(x):\n",
```

Comment:
What does x represent? It's better to use variable names that make it clear what the variable contains